checks/mixins/container_engine_cpe.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2024 Swiss National Supercomputing Centre (CSCS/ETH Zurich)
+# ReFrame Project Developers. See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+import reframe as rfm
+import reframe.utility.typecheck as typ
+
+
+class ContainerEngineMixinCPE(rfm.RegressionMixin):
```

Comment:
As discussed on slack, I am not sure we need to have a special class for this. I think we can re-use the container engine mixin and make pass the options in both compilation and run scripts